src/sentry/integrations/gitlab/issues.py

```diff
@@ -25,9 +25,14 @@ def get_persisted_default_config_fields(self) -> Sequence[str]:
     def get_projects_and_default(self, group, **kwargs):
         params = kwargs.get("params", {})
         defaults = self.get_project_defaults(group.project_id)
-        kwargs["repo"] = params.get("project", defaults.get("project"))
 
-        # In GitLab Repositories are called Projects
+        # XXX: In GitLab repositories are called projects but get_repository_choices
+        # expects the param to be called 'repo', so we need to rename it here.
+        if not params:
+            # Ensure kwargs has a reference to the params dict so 'repo' is accessible when kwargs is passed
+            kwargs["params"] = params
+        params["repo"] = params.get("project", defaults.get("project"))
```

Comment:
🤔 

So - you're setting the reference to the params dictionary
then you're updating the params dictionary?

this feels confusing...

Reply:
Oh oof, and params is coming from kwargs in initially too???

Can we just update this entire method?
```
def get_projects_and_default(self, group, params = {}, **kwargs):
    defaults = self.get_project_defaults(group.project_id)
    params['repo'] = params.get('project') or defaults.get('project')
    default_project, project_choices = self.get_repository_choices(group, params, **kwargs)
    
    return default_project, project_choices
```

kwargs doesn't need to be magically constructed/deconstructed like you're doing it here.
Looks like you only really care about the `params` kwarg, so lets just pull that out and deal with it accordingly

Reply:
the `or` rather than defaulting the `get` just feels more readable to me too

Reply:
Changed it to:
```Python
def get_projects_and_default(self, group: Group, params: dict, **kwargs):
def get_repository_choices(self, group: Group, params: dict, **kwargs):
```

Then for Gitlab where we call `get_projects_and_default`, I pull out the param doing `kwargs.get("params", {}).
The other integrations don't have that helper, so they directly call `get_repository_choices`. I do the same thing and pull out `params`